### PR TITLE
refactor: permissions v2

### PR DESCRIPTION
**Pull request by vitorlops**
**Please describe the changes this PR makes and why it should be merged:**

**If applicable, please reference Discord API Docs PRs or commits that influenced this PR:**
- https://github.com/discord/discord-api-docs/pull/4830

### CODE REVIEW

File: deno/payloads/v10/oauth2.ts [hunk 1]
```diff
@@ -111,4 +111,10 @@ export enum OAuth2Scopes {
 	 * See https://discord.com/developers/docs/interactions/application-commands
 	 */
 	ApplicationsCommandsUpdate = 'applications.commands.update',
+	/**
+	 * Allows your app to update permissions for its Application Commands using a bearer token - client credentials grant only
```
Comment (vladfrangu) [inline deno/payloads/v10/oauth2.ts:115]: This is not client credentials only 🤔

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: deno/payloads/v10/oauth2.ts (hunk 1 above)
Comment (almeidx) [reply deno/payloads/v10/oauth2.ts:115]: https://github.com/discord/discord-api-docs/pull/4830/files#diff-d557cf0682f63e6ebb8f6646617ea3c17e359a5fc0f204e99a6980642d785edaR32

Comment (vitorlops) [reply deno/payloads/v10/oauth2.ts:115]: Well, I just copied and formated from the upstream repository.
image

File: deno/payloads/v10/_interactions/_applicationCommands/permissions.ts [hunk 2]
```diff
@@ -40,10 +40,16 @@ export interface APIApplicationCommandPermission {
 	permission: boolean;
 }
 
+export enum ApplicationCommandPermissionsConstant {
+	Everyone = 'guild_id',
```
Comment (vladfrangu) [inline deno/payloads/v10/_interactions/_applicationCommands/permissions.ts:44]: This is not useful for users at all. Lets make this an object where we take in a bigint|string, and return the expected value back  (also no, you don't pass in the literal `guild_id` string :D

**Review by vladfrangu — CHANGES_REQUESTED** [review]

**Comment by vitorlops** [comment]
Actually, I don't think I could make all this stuff by myself, since I don't have the correct skills (and time).
I'm sorry for maybe disturbing someone but I'm going to close it.

**Pull request closed without merging**